### PR TITLE
Bugfix/issue 830 support kotlin 1 4

### PR DESCRIPTION
**Pull request by Vakhturov**
Updated Kotlin plugin to v1.4.0 and Gradle to v.6.6

PS: contributing for first time here, any feedback appreciated.
Couldn't publish to maven local: 
`Invalid publication 'maven': multiple artifacts with the identical extension and classifier ('jar.asc', 'all')`
Is it be because of the verification-metadata.xml?

### CODE REVIEW

**Review by henrik242 — APPROVED** [review]
lgtm

**Comment by Vakhturov** [comment]
closed unmerged by mistake

File: build.gradle
Comment (eduardompinto) [inline build.gradle:7]: Since there's one stable release. Does it make sense keeping this release candidate here?

File: settings.gradle
Comment (eduardompinto) [inline settings.gradle:11]: Also here.

**Comment by eduardompinto** [comment]
Also Kotlin 1.4 is already handled by the version https://github.com/pinterest/ktlint/releases/tag/0.38.0-alpha01.
I think the will release a non-alpha 0.38.0 into the next days.

File: settings.gradle
Comment (Vakhturov) [reply settings.gradle:11]: Also had this idea (see first commit), but found this in readme file

> Additionally, project publishes snapshots build against latest kotlin development version. To use them, change version
> of ktlint to `<latest-version>-kotlin-dev-SNAPSHOT`.
I guess, this is added to test ktlint against any other Kotlin dev version.

**Comment by Vakhturov** [comment]
@eduardompinto thanks for comments. We were kinda blocked by this issue in the project, that's why I wanted to contribute for the latest Kotlin support. 
Would be happy if this PR helps. Also fine if anyone else is already about to release it.

**Comment by Vakhturov** [comment]
@eduardompinto Could find the changeset with updated Kotlin plugin to 1.4.0. Found only this branch.
https://github.com/pinterest/ktlint/compare/0.37.2...0.38.0-alpha01

**Comment by eduardompinto** [comment]
I think one of the project maintainers can talk about the release date, but the alpha version is already available in the maven repository
https://search.maven.org/artifact/com.pinterest/ktlint/0.38.0-alpha01/jar

**Comment by shashachu** [comment]
Thanks for the PR @Vakhturov. I'm actually likely going to close this in favor of another upcoming diff; myself and the other maintainers just have to chat a little bit about ongoing preview Kotlin versions. 

In general, we'd also prefer diffs that do one thing, e.g. upgrade Kotlin OR Gradle, but not both. 

And like @eduardompinto said, you can use the alpha01 version if you need to be immediately unblocked. 

Looking forward to your future contributions to ktlint. Thanks!

**Comment by Vakhturov** [comment]
@shashachu thanks for the feedback. Looking forward for the upcoming release!

**Pull request closed without merging**
